### DISCUSSION TITLE
Different alignment for top and bot

### DISCUSSION THREAD

**Discussion by cu1ch3n** [typst/typst#3827] [General]
### Description

MWE:
```typst
$1 + top + 1$ \
$1 + bot + 1$
```
![](https://github.com/typst/typst/assets/80438676/2ddc4ab6-985e-4a50-bf72-7049356f204e)

### Reproduction URL

https://typst.app/project/rK-1vDa1XN7tJsfp_cQkAN

### Operating system

Web app

### Typst version

- [X] I am using the latest version of Typst

**Comment by Enivex** [comment]
This is not a bug,

https://www.unicode.org/Public/math/revision-15/MathClass-15.txt

U+22A4 (top) is of class "normal" while U+22A5 (bot) is of class "relation".

You can however force typst to consider it to be normal by using `class("normal", bot)`

```typ
$1 + top + 1$ \
$1 + class("normal",bot) + 1$
```

![image](https://github.com/typst/typst/assets/2191754/a9ebd67b-2c11-4ee3-88ad-f3e6614815f0)

**Reply by chengluyu** [reply]
I think this is a wrong classification, because we have two similar symbols in Unicode: [up tack ⊥](https://www.compart.com/en/unicode/U+22A5) and [perpendicular ⟂](https://www.compart.com/en/unicode/U+27C2). Even though they look the same, their uses are completely different. The up tack is used in type theory to represent the [bottom type](https://en.wikipedia.org/wiki/Bottom_type), which is a type with no values. We should differentiate them in Typst, otherwise it would cause trouble for different usages.

**Comment by cu1ch3n** [comment]
It is weird to me. I think `bot` (`\bot` in LaTex, `U+22A5` in Unicode) is normally a constant symbol, while the similar-looking perpendicular symbol (⟂, `\perp` in LaTeX, `U+27C2` in Unicode) is a acutally binary relation symbol. See https://en.wikipedia.org/wiki/Up_tack .

In LaTex:
```latex
\begin{align*}
1 + \bot + 1 \\
1 + \top + 1 \\
1 + \perp + 1
\end{align*}
```

The result would be:

![](https://github.com/typst/typst/assets/80438676/e6038f84-6e4d-4591-92a9-c71d6ba0511c)


I think people would regard `bot` as the constant. Maybe we can covert it into normal class and keep `perp` as a binary relation?

**Reply by Enivex** [reply]
It may be weird to you, but I just sent you a link to the list of classifications from the Unicode consortium. It's not something typst came up with.

Good news is that it's very easy to redefine `#let bot =$class("normal",bot)$`

**Reply by cu1ch3n** [reply]
Thanks for the reply. But if I write `$⊥$` instead of `$bot$`, the workaround doesn't work...

**Reply by cu1ch3n** [reply]
`#show math.bot: bot` should work

**Comment by cu1ch3n** [comment]
not sure if it is related: https://en.wikipedia.org/wiki/List_of_XML_and_HTML_character_entity_references#cite_note-perp-31

> &perp;: Unicode only defines U+22A5 as the "up tack", and the Unicode symbol for "perpendicular" is U+27C2: the two symbols look similar, but are separate in Unicode. However, HTML uses U+22A5 as its "perpendicular" symbol: this is a discrepancy between HTML and Unicode. As well, the U+22A4 character (the "down tack" symbol) rendered in a browser such as [Firefox](https://en.wikipedia.org/wiki/Firefox) 3.6 can match the [font](https://en.wikipedia.org/wiki/Font) of either "up tack" or "perpendicular", but not both, depending on whether a fixed-width or a proportional font is used. When viewed in Firefox 3.6, the symbols rendered in the order U+22A5, U+22A4, U+27C2 in a proportional font: "⊥ ⊤ ⟂" and a fixed width one: ⊥ ⊤ ⟂, shows that the "down tack" has a similar look to U+22A5 (HTML's "perpendicular") in the first case but matches U+27C2 in the second. This exemplifies the difficulties of the [semiotics](https://en.wikipedia.org/wiki/Semiotics) involved in interpreting [glyphs](https://en.wikipedia.org/wiki/Glyphs), [symbols](https://en.wikipedia.org/wiki/Symbols) and [characters](https://en.wikipedia.org/wiki/Character_(symbol)) generally.
